### PR TITLE
chore(actions): check uncomitted go source code changes

### PR DESCRIPTION
**Pull request by JGAntunes**
This picks up on what has been discussed on #285 and adds a step to verify that the go client is up to date and the files have been checked in (once again based on work @erezrokah did in the CLI 🙏 ). I've tested it in this PR by making some changes to  `swagger.yml` file and not generating a new client - https://github.com/netlify/open-api/runs/1742899676?check_suite_focus=true. - seems solid 😄

I've also added two files to the `.gitignore`, which I believe are only used as [helpers when generating the go glient](https://github.com/netlify/open-api/blob/master/generate.go#L4) and don't need to be checked in. Do tell me if that is not the case though.

### CODE REVIEW

**Review by keiko713 — APPROVED** [review]
I remember that @erezrokah and I were talking about these swagger_*.json files the other day for what to do. I think it's safe to add them to `.gitignore`, so looks good! 👍

File: .github/workflows/verify-go-src.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  pull_request:
+    branches: [master]
+
+jobs:
+  verify-go:
+    name: Verify go client source code is up to date
+    strategy:
+      matrix:
+        os: [ubuntu-latest]
+        go_version: [1.14.x]
```
Comment (mraerino) [inline .github/workflows/verify-go-src.yml:15]: latest is `1.15`, do we want to use that?

**Review by mraerino — APPROVED** [review]
awesome!

File: .github/workflows/verify-go-src.yml (hunk 1 above)
Comment (JGAntunes) [reply .github/workflows/verify-go-src.yml:15]: Happy to do it, but maybe in that case we should also add it to our `Test` action - https://github.com/netlify/open-api/blob/master/.github/workflows/test.yml#L34?

**Pull request merged**